server/src/main/java/org/opensearch/index/store/RemoteSegmentStoreDirectory.java

```diff
@@ -408,6 +447,20 @@ public Map<String, UploadedSegmentMetadata> getSegmentsUploadedToRemoteStore() {
         return Collections.unmodifiableMap(this.segmentsUploadedToRemoteStore);
     }
 
+    public Map<String, UploadedSegmentMetadata> getSegmentsUploadedToRemoteStore(long primaryTerm, long generation)
```

Comment:
These methods which are not part of Directory interface should be part of some wrapper/util for `RemoteSegmentStoreDirectory` and can be static.

Reply:
makes sense. will update them